src/main/java/org/linksharing/server/auth/AuthController.java

```diff
@@ -40,6 +41,10 @@ public String showRegistrationPage(User user) {
     @PostMapping("/register")
     public String registerUser(@Valid User user, BindingResult result) {
```

Comment:
The logic in this method doesn't look right...

Reply:
For example, you call `existsByUsername` twice.

Reply:
Not sure that got you right, but do admit - method was raw. Fixed it. Please review.